pkg/controller/database/postgresqlserver/managed.go

```diff
@@ -154,7 +174,13 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 
 	cr.SetConditions(xpv1.Creating())
 
-	pw, err := e.newPasswordFn()
+	pw, err := e.getPassword(ctx, cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
```

Comment:
I thought you said it'd save the password to the connection secret before triggering the creation. Is that not the case anymore?

Reply:
Nope, it was never the case (also not the case when I ran the above set of experiments). We still leave it to the managed reconciler.

Reply:
Got it. So, we save the password after a successful `CreateServer` call via the managed reconciler and if `CreateServer` is ever called again, we'd be getting the password from connection secret.